### PR TITLE
test: expand test coverage with mutation-killing assertions

### PR DESCRIPTION
**Pull request by otomatty**
## 概要

大幅にテストカバレッジを拡張し、境界値、エラーハンドリング、状態遷移、キー再利用などの細かなロジックを検証するテストを追加しました。各テストは特定の実装詳細（例：正規表現の `+` vs `*`、比較演算子の `<` vs `<=`、配列スライスのオフセット）を狙った変異テストキラーとして設計されています。

### `src/hooks/useImageUpload.test.ts`
- **モック構造の改善**: ホイスト済みモック状態コンテナを導入し、各テストがモジュール再定義ではなく状態を直接変更できるように改善
- **ヘルパー関数追加**: `makeFile()` で簡潔にテスト用ファイルを生成
- **uploadImage メソッドの包括的テスト**:
  - AbortSignal の伝播、戻り値の検証
  - 既に abort 済みの signal での同期的エラー
  - ストレージ未設定時のエラーメッセージ固定
  - 非画像ファイル型の拒否（`image/` 判定の変異を殺す）
  - JPEG/PNG の WebP 変換、GIF/WebP/SVG の非変換（条件分岐の反転変異を殺す）
  - onProgress コールバックの配線検証
  - エラーハンドリング（Error vs 非Error、abort vs 通常エラー）
  - 成功時の終端状態（isUploading=false、progress=100%）
- **uploadImages メソッドの包括的テスト**:
  - 非画像ファイルのフィルタリング
  - フィルタ後に画像が 0 件の場合のエラー
  - 入力順の保持（Promise.all の順序保証）
  - バッチエラーハンドリング
- **isConfigured プロパティのテスト**:
  - ロード中の false、未設定時の false、設定済みの true
- **clearError メソッドのテスト**:
  - error のみリセット、他の状態は不変

### `src/lib/encryption.test.ts`
- **テスト構造の再編成**: 「round-trip」「IV semantics」「authentication」「key persistence」「clearEncryptionKey」の 5 つのカテゴリに分類
- **round-trip テスト**:
  - 空文字列の暗号化・復号（off-by-one スライス変異を殺す）
  - Unicode/multibyte テキストの検証
- **IV セマンティクスの検証**:
  - 毎回異なる IV 生成（`crypto.getRandomValues` の削除変異を殺す）
  - IV が先頭に配置されていることの直接検証（`combined.set()` のオフセット変異を殺す）
  - IV 改竄時の復号失敗
  - 復号時の IV 読み込み位置検証（スライス引数の変異を殺す）
- **認証・改竄耐性**:
  - 暗号化データ部の改竄検出（GCM タグ検証）
- **キー永続化と再利用**:
  - 初回 encrypt での localStorage 保存
  - 2 回目以降の encrypt での既存キー再利用（新規

https://claude.ai/code/session_012VD4wm7gStZmNZ2tf4gST9

---

<a href="https://app.devin.ai/review/otomatty/zedi/pull/740" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

## Summary by CodeRabbit

* **Tests**
  * Expanded and reorganized test suites across image uploads, conversation title generation, encryption, and argument parsing.
  * Image upload tests: abort/timeout handling, progress, MIME/type-specific conversion rules, error states, provider passthrough, and multi-file flows.
  * Title generation: selection rules, truncation (including multibyte), and edge cases.
  * Encryption: IV placement, tamper resistance, key persistence/clearing, and round-trip cases.
  * Argument parsing: whitespace, quoting, escapes, and mixed-token behaviors; added reusable test helpers.

### CODE REVIEW

File: src/lib/encryption.test.ts [hunk 1]
```diff
@@ -2,47 +2,212 @@ import { describe, it, expect, beforeEach } from "vitest";
 import { encrypt, decrypt, clearEncryptionKey } from "./encryption";
 
 const hasSubtle = typeof globalThis.crypto?.subtle !== "undefined";
+const ENCRYPTION_KEY_NAME = "zedi-encryption-key";
```
[the remark this answers is not in the document]
Comment (otomatty) [reply src/lib/encryption.test.ts:5]: 対応不要 / Not actionable — 実装と一致しています (false positive)。

`src/lib/encryption.ts:9` の定数も `"zedi-encryption-key"` (ハイフン) を使用しています:

```ts
// src/lib/encryption.ts
const ENCRYPTION_KEY_NAME = "zedi-encryption-key";
```

```bash
$ grep -n "ENCRYPTION_KEY_NAME\|zedi_encryption_key\|zedi-encryption-key" src/lib/encryption.ts
9:const ENCRYPTION_KEY_NAME = "zedi-encryption-key";
15:  const storedKey = localStorage.getItem(ENCRYPTION_KEY_NAME);
38:  localStorage.setItem(ENCRYPTION_KEY_NAME, keyBase64);
97:  localStorage.removeItem(ENCRYPTION_KEY_NAME);
```

リポジトリ全体に `zedi_encryption_key` (アンダースコア) は存在しません。`stores the generated key in localStorage on first encrypt` などキー永続化系のテストは実際に通過しており（`encrypt()` 後に同じキー名で localStorage から取得できることを確認している）、文字列が不一致であれば成立しません。


---
_Generated by [Claude Code](https://claude.ai/code)_
